### ISSUE TITLE
Improve build.cmd to detect hanging processes that are known to disrupt a proper build

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#10274]
As described in [the testguide.md](https://github.com/dotnet/fsharp/blob/main/TESTGUIDE.md#error-that-a-file-cannot-be-accessed), if developers work with both `build.cmd` and Visual Studio (`devenv.exe`), conflicts arise when VS is not closed or VS (or the build run from VS) doesn't properly close or kill its processes upon exiting VS.

Just now I had a discussion on Slack with @laenas where `build.cmd` repeated weird errors. Closing VS did most of the trick (we're working on the rest). Moral here is: I think it shouldn't be too hard to have a "before all" command in the `build.cmd` that checks for _known suspect processes_ and warns about them and then quits.

Perhaps with an option to ignore said warning, or another option to kill the processes if they are known to be abandoned (i.e., have no parent process), but this can lead to lost work.

Basically just reporting this as a reminder to fix sometime, after coming across it time and time again. Not an actual bug though.

**Comment by cartermp** [comment]
Triaging as a contributor pain issue; I don't think we'd do this work but we might welcome a PR that looks at some of these processes.

**Comment by abelbraaksma** [comment]
Yeah, I wanted to tackle this a few times, just kept forgetting ;). An alternative approach is to catch/filter for "file in use" errors and show a meaningful description like "One or more files are in use, make sure you close processes so and so" at the end. Currently such messages can be deeply buried and the build doesn't immediately quit.

I believe I have a few other issues open with related improvements. Nothing major, but may certainly help first-time contributors if this gets improved.
